### PR TITLE
Validations

### PR DESCRIPTION
**Pull request by redwing555**
In this milestone, the following is done : 

- Adding the following validations:

 -  For the User model:

    1. Name must not be blank.
    2. PostsCounter must be an integer greater than or equal to zero.

 -  For the Post model:

    1. Title must not be blank.
    2. Title must not exceed 250 characters.
    3. CommentsCounter must be an integer greater than or equal to zero.
    4. LikesCounter must be an integer greater than or equal to zero.

- Adding unit specs for all of the models' methods and validations.
- Adding flash messages in the create actions was done in the previous PR.
- Making sure that the N+1 problem is solved when fetching all posts and their comments for a user.

### CODE REVIEW

File: app/controllers/posts_controller.rb [hunk 1]
```diff
@@ -1,12 +1,11 @@
...
   def show
     @user = User.find_by_id(params[:user_id])
-    @post = Post.find_by_id(params[:id])
+    @post = @user.posts.includes(:likes, :comments).find(params[:id])
   end
 
   def index
-    @user = User.find(params[:user_id])
-    @posts = @user.posts.order(created_at: :desc)
+    @user = User.includes(:posts).find(params[:user_id])
```
Comment (AshakaE) [inline app/controllers/posts_controller.rb:8]: - [x] Kindly add the comments in the index method, so posts come along with the comments

**Review by AshakaE — CHANGES_REQUESTED** [review]
### Hi @redwing555  🙋‍♀️, 

Good job so far! 👌
There are some issues that you still need to work on to go to the next project but you are almost there!
<FEEL FREE TO CHANGE THIS MESSAGE TO YOUR PERSONAL WORDS>

- [x] Please use flash messages for both successful and unsuccessful resource creation(posts, likes and comments).

#  Required Changes ♻️ 🚧
_Check the comments under the review._

###  Optional suggestions
_Every comment with the [OPTIONAL] prefix is not crucial enough to stop the approval of this PR. However, I strongly recommend you to take them into account as they can make your code better._


Cheers and Happy coding!👏👏👏

Feel free to leave any questions or comments in the PR thread if something is not 100% clear.

**_Please, do not open a new Pull Request for re-reviews. You should use the same Pull Request submitted for the first review, either valid or invalid unless it is requested otherwise._**

------
_As described in the [Code reviews limits policy](https://microverse.zendesk.com/hc/en-us/articles/1500004088561) you have a limited number of reviews per project (check the exact number in your Dashboard). If you think that the code review was not fair, you can request a second opinion using [this form](https://airtable.com/shrQAqnBwek5a0O0s)._

**Comment by redwing555** [comment]
Hello @AshakaE, 

Thank for the review, I have added flash messages for posts, likes, and comments in the previous PR, I lacked an unsuccessful comment creation action that I have added now, I also added the comments in the index controller as requested although I don't use it in the post index view.

Thank you and happy coding

**Review by Dipeshtwis — APPROVED** [review]
# Status: Approved ✔️

## Hi @redwing555, 

Good job on implementing the suggested changes. Well Done.
Your project is complete!  There is nothing else to say other than... it's time to merge it :shipit: 
Congratulations! 🎉 

**You may close this Pull Request and merge the changes**

  
**Keep being awesome devs and good luck with your next milestone!**
---------

Cheers and Happy coding!👏👏👏
Feel free to leave any questions or comments in the PR thread if something is not 100% clear.
_**Dipesh Kumar**_

--------

_As described in the [Code reviews limits policy](https://microverse.zendesk.com/hc/en-us/articles/1500004088561) you have a limited number of reviews per project(check the exact number in your Dashboard). If you think that the code review was not fair, you can request a second opinion using [this form](https://airtable.com/shrQAqnBwek5a0O0s)._

**Comment by redwing555** [comment]
Hello @Dipeshtwis,

Thanks for the review, happy coding mate! 

Cheers 🥂

**Pull request merged**